src/dymoprint/dymo_print_engines.py

```diff
@@ -145,38 +145,33 @@ def render_barcode_with_text(
         assert align in ("left", "center", "right")
         # Generate barcode
         code_bitmap = self.render_barcode(barcode_input_text, bar_code_type)
-        code_bitmap = code_bitmap.resize(
-            (code_bitmap.width * 2 // 3, code_bitmap.height * 2 // 3)
-        )
 
         # Generate text
         text_bitmap = self.render_text(
-            barcode_input_text, font_file_name, frame_width, font_size_ratio, align
-        )
-        text_width = int(
-            text_bitmap.width // (text_bitmap.height / (code_bitmap.height / 2))
+            text_lines=barcode_input_text,
+            font_file_name=font_file_name,
+            frame_width_px=frame_width,
+            font_size_ratio=font_size_ratio,
+            align=align,
+            label_height_px=code_bitmap.height // 3,
```

Comment:
I added the argument names to improve readability. I also added a new `label_height_px` so that we can resize the text within the `render_text` function.